js/components/ManuscriptForm.js

```diff
@@ -1,43 +1,116 @@
+ * selection form (aka the scriptchart configuration form).
+ * Changes made to any of the form options eventually make their way
+ * back to this component's state.
+ * Its state is sent back to the App as the form data, to be
+ * processed by the DashTabs and ScriptChart components when the
+ * form is submited.
+ */
+
 import ManuscriptMenu from "./ManuscriptMenu";
 import LettersLoader from "./LettersLoader";
 
-// In this component, we would actually pass a list of
-// manuscripts, and have a toggle to sort by date or name
-// The list could come from a json object with name and date
-
 class ManuscriptForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    /* The form defaults are largely set here */
+    this.state = {
+      showBinarized: true,
+      letterExamples: 3,
+      imageSize: "Small",
+      selectedShelfmarks: [],
+      manuscripts: [],
+      letters: []
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSelect = this.handleSelect.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    const target = event.target;
+    const name = target.name;
+    const value = target.type === "checkbox" ? target.checked : target.value;
+
+    this.setState({
+      [name]: value
+    });
+  }
+
+  /* This is the handler for the multi-select items (manuscript list and letter
+   * button grid). It would be nice to integrate this with handleChagne(),
+   * but that hasn't been workable so far.
+   */
+  handleSelect(name, value) {
+    this.setState({
+      [name]: value
+    });
+  }
+
+  handleSubmit(event) {
+    // Stop the whole darn page from reloading on submit
+    event.preventDefault();
+    // Pass all of the form's state to the handler (which is in DashTabs)
+    this.props.formSubmit(this.state);
+  }
+
   render() {
+    console.log("Rendering ManuscriptForm");
     return (
-      <form className={"manuscript-form"}>
-        <ManuscriptMenu />
+      <form className={"manuscript-form"} onSubmit={this.handleSubmit}>
+        <ManuscriptMenu
+          handleChange={this.handleChange}
+          handleSelect={this.handleSelect}
+          manuscripts={this.props.manuscripts}
+        />
         <div className={"field"}>
           <label className={"control"}>Select Letters:</label>
-          <LettersLoader />
+          <LettersLoader handleSelect={this.handleSelect} />
         </div>
         <div className={"field"}>
           <label className={"control"}>Select number of letter examples:</label>
           <div className={"control"}>
             <div className={"select"}>
-              <select>
+              <select
+                value={this.state.letterExamples}
+                type="number"
+                name="letterExamples"
+                onChange={this.handleChange}
+              >
                 <option>1</option>
                 <option>2</option>
```

Comment:
I think we had originally talked about just having 1, 3, 5 as the options for number of images. Doesn't hurt to have the extras, but it may not be necessary.

Reply:
I think it's preferable to pare down the list a bit, so let's just do it and see if it prompts any feedback.